learn/react-native/device-biometric-authentication.md

```diff
@@ -0,0 +1,122 @@
+- Authenticate the user with biometric authentication.
+
+## Adding LocalAuthentication to your WaveMaker React Native App
+
+LocalAuthentication plugin can be installed in few steps in WaveMaker. Please refer to this [page](https://docs.wavemaker.com/learn/react-native/third-party-expo-plugins#expo) 
+on how to install a plugin.
+
+Once the plugin is added, you'll be seeing something similar to this in your screen.
+
+![LocalAuthentication](/learn/assets/install-localauth.png)
+
+## How to list supported local authentication
+
+LocalAuthentication plugin provides an API `supportedAuthenticationTypeAsync()` to list supported authentication types available in a device.
+This API returns a promise that resolves to an array of strings, each of which is a constant defined in the [`AuthenticationType`](https://docs.expo.dev/versions/latest/sdk/local-authentication/#authenticationtype) enum.
+
+## How to invoke any local authentication
+
+'authenticateAsync()' API is used to invoke any local authentication. This API returns a promise that resolves to a boolean value indicating whether or not the authentication was successful.
+
+```javascript
+Page.authenticateTap = async function($event, widget) {
+    var result = '';
+    try {
+        const results = await LocalAuthentication.authenticateAsync();
+        if (results.success) {
+            Page.msgerror = false;
+            result = 'Successfully authenticated';
+        } else if (results.error === 'unknown') {
+            Page.msgerror = true;
+            result = 'Biometric authentication has been disabled';
+        } else if (
+            results.error === 'user_cancel' ||
+            results.error === 'system_cancel' ||
+            results.error === 'app_cancel'
+        ) {
+            Page.msgerror = true;
+            result = 'Authentication process has been cancelled';
+        } else if (results.error === 'not_available') {
+            Page.msg_error = true;
+            result = 'BioMetric Authentication is not enabled';
+        }
+    } catch (error) {
+        Page.msgerror = true;
+        result = 'There was an error in authentication';
+    }
+    Page.Widgets.message.show = true;
+    Page.Widgets.resetauth.show = true;
+    Page.Widgets.authenticate.show = false;
+    Page.Widgets.message.caption = result;
+    Page.refresh();
+};
+```
+
+## How to invoke only Biometric authentication 
+
+We can restrict the authentication to only biometric authentication by passing the [`options`](https://docs.expo.dev/versions/latest/sdk/local-authentication/#localauthenticationauthenticateasyncoptions) parameter to the `authenticateAsync()` API.
+
+- `disableDeviceFallback` -- If set to true, the user will be unable to authenticate with biometrics (Touch ID or Face ID) and will be asked for their passcode instead. Defaults to false.
+- `requireConfirmation` -- Whether to show a dialog after successful biometric authentication. Defaults to true on Android.
```

Comment:
:::note
`requireConfirmation` applies to Android only, and the default value sets to true. 
:::

Reply:
updated